extensions/profiler/client/src/data/createSessionData.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as sqlops from 'sqlops';
+
+export class CreateSessionData {
+	public ownerUri: string;
+	public sessionName: string;
+	public templates: Array<sqlops.ProfilerSessionTemplate> = new Array<sqlops.ProfilerSessionTemplate>();
+
+	constructor(ownerUri: string, templates: Array<sqlops.ProfilerSessionTemplate>) {
+		this.ownerUri = ownerUri;
+		this.templates = templates;
+	}
+
+	public getTemplateNames(): string[] {
+		return this.templates.reduce<Array<string>>((p, e) => {
```

Comment:
Simpler code for the same thing would be `return this.templates.map(e => e.name)`